File: 1071-greatest-common-divisor-of-strings/1071-greatest-common-divisor-of-strings.cpp
class Solution {
    int gcd(int a, int b) {
    while (b != 0) {
        int temp = b;
        b = a % b;
        a = temp;
    }
    return a;
}
public:
    string gcdOfStrings(string str1, string str2) {
        size_t index = 0;

        if (str1 + str2 != str2 + str1)
            return "";

        index = gcd(str1.length(), str2.length());

        return str1.substr(0, index);;
    }
};

File: 1071-greatest-common-divisor-of-strings/1071-greatest-common-divisor-of-strings.ts
function gcdOfStrings(str1: string, str2: string): string {
    function gcd(a: number, b: number): number {
        while (b != 0) {
            const temp: number = b;
            b = a % b;
            a = temp;
        }
        return a;
    }
    if (str1 + str2 == str2 + str1) {
        return str1.substring(0, gcd(str1.length, str2.length));
    } return "";
};